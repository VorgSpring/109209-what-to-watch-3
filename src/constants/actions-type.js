export const CHANGE_GENRE = `CHANGE_GENRE`;

export const LOAD_FILMS_REQUEST = `LOAD_FILMS_REQUEST`;
export const LOAD_FILMS_SUCCESS = `LOAD_FILMS_SUCCESS`;
export const LOAD_FILMS_ERROR = `LOAD_FILMS_ERROR`;

export const AUTHORIZATION_REQUEST = `AUTHORIZATION_REQUEST`;
export const AUTHORIZATION_SUCCESS = `AUTHORIZATION_SUCCESS`;
export const AUTHORIZATION_ERROR = `AUTHORIZATION_ERROR`;

export const AUTHENTICATED_USER = `AUTHENTICATED_USER`;

export const CHANGE_FAVORITES_REQUEST = `CHANGE_FAVORITES_REQUEST`;
export const CHANGE_FAVORITES_SUCCESS = `CHANGE_FAVORITES_SUCCESS`;
export const CHANGE_FAVORITES_ERROR = `CHANGE_FAVORITES_ERROR`;

export const LOAD_PROMO_MOVIE_REQUEST = `LOAD_PROMO_MOVIE_REQUEST`;
export const LOAD_PROMO_MOVIE_SUCCESS = `LOAD_PROMO_MOVIE_SUCCESS`;
export const LOAD_PROMO_MOVIE_ERROR = `LOAD_PROMO_MOVIE_ERROR`;

export const LOAD_REVIEWS_REQUEST = `LOAD_REVIEWS_REQUEST`;
export const LOAD_REVIEWS_SUCCESS = `LOAD_REVIEWS_SUCCESS`;
export const LOAD_REVIEWS_ERROR = `LOAD_REVIEWS_ERROR`;
