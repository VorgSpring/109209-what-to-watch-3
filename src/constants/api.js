export const ApiPaths = {
  BASE_URL: `https://htmlacademy-react-2.appspot.com/wtw`,
  FILMS: `/films`,
  PROMO_MOVIE: `/films/promo`,
  LOGIN: `/login`,
  setFavorite({id, status}) {
    return `/favorite/${id}/${status}`;
  },
  getReviews(id) {
    return `/comments/${id}`;
  }
};

export const ApiConfiguration = {
  baseURL: ApiPaths.BASE_URL,
  timeout: 5000,
  withCredentials: true
};

export const ResponseStatus = {
  UNAUTHORIZED: 401
};
