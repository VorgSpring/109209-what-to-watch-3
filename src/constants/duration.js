export const PLAY_PREVIEW_DURATION = 1000; // 1сек
