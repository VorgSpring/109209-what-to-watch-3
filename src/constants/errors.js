import {FormFields} from './form-fields';

// TODO описать нормально ошибки
export const FormFieldsErrors = {
  [FormFields.EMAIL]: `некорректный email`,
  [FormFields.PASSWORD]: `некорректный password`,
  [FormFields.REVIEW]: `текст отзыва должен быть не меньше 50 и не больше 400 символов`,
  [FormFields.RATING]: `выставите оценку фильму`
};

export const ServerErrors = {
  INTERNAL_ERROR: `ошибка сервера`
};

export const RequestErrors = {
  NOT_FOUND: `сервер не найден`,
  BAD_REQUEST: `некорректный запрос`,
  UNAUTHORIZED: `ошибка авторизации`
};
