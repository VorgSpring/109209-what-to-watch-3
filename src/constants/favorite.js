export const FavoriteStatus = {
  ADD: 1,
  REMOTE: 0
};

export const IconFavoriteButton = {
  ADD: `#add`,
  IN_LIST: `#in-list`
};
