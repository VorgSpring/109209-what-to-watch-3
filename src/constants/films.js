export const MAX_COUNT_SIMILAR_FILMS = 4;

export const ALL_GENRES_TYPE = `All genres`;
