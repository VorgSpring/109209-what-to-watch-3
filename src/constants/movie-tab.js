import {MovieOverview} from '../components/movie-overview/movie-overview.jsx';
import {MovieDetails} from '../components/movie-details/movie-details.jsx';
import {MovieReviewsContainer} from '../components/movie-reviews/movie-reviews.jsx';

export const MovieTabItems = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};

export const MovieTabComponents = {
  [MovieTabItems.OVERVIEW]: MovieOverview,
  [MovieTabItems.DETAILS]: MovieDetails,
  [MovieTabItems.REVIEWS]: MovieReviewsContainer
};
