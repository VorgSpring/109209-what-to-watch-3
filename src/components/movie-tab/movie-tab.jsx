import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {MovieTabItems} from '../../constants/movie-tab';

export class MovieTab extends PureComponent {
  constructor(props) {
    super(props);

    this.setActiveTab = this.setActiveTab.bind(this);
  }

  setActiveTab() {
    const {
      isActive,
      name,
      onActive
    } = this.props;

    if (!isActive) {
      onActive(name);
    }
  }

  render() {
    const {
      isActive,
      name
    } = this.props;

    return (
      <li
        className={`movie-nav__item ${isActive && `movie-nav__item--active`}`}
        onClick={this.setActiveTab}
      >
        <span className='movie-nav__link'>
          {name}
        </span>
      </li>
    );
  }
}

MovieTab.propTypes = {
  isActive: PropTypes.bool.isRequired,
  name: PropTypes.oneOf(
      Object.values(MovieTabItems)
  ).isRequired,
  onActive: PropTypes.func.isRequired
};
