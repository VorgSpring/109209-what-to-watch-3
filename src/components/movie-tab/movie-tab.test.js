import React from 'react';
import {MovieTab} from './movie-tab.jsx';
import renderer from 'react-test-renderer';
import {MovieTabItems} from '../../constants/movie-tab';

describe(`MovieTab`, () => {
  it(`should renders correctly`, () => {
    const tree = renderer.create(
        <MovieTab
          isActive={true}
          name={MovieTabItems.DETAILS}
          onActive={()=>{}}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
