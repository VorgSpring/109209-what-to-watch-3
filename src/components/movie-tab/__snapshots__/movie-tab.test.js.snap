// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTab should renders correctly 1`] = `
<li
  className="movie-nav__item movie-nav__item--active"
  onClick={[Function]}
>
  <span
    className="movie-nav__link"
  >
    Details
  </span>
</li>
`;
